common/functions/src/scalars/dates/to_interval_function.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use common_datavalues::prelude::*;
+use common_datavalues::IntervalKind;
+use common_datavalues::IntervalType;
+use common_exception::Result;
+
+use crate::scalars::Function;
+use crate::scalars::FunctionContext;
+use crate::scalars::FunctionDescription;
+use crate::scalars::FunctionFeatures;
+
+macro_rules! impl_to_interval_function {
+    ($FUNCTION_NAME:ident, $INTERVAL_KIND: ident) => {
+        #[derive(Clone)]
+        pub struct $FUNCTION_NAME {
+            display_name: String,
+        }
+
+        impl $FUNCTION_NAME {
+            pub fn try_create(
```

Comment:
Wow, I don't think it's good practice to use macros here.

Why not use :

```
IntervalFunction {
   pub try_create(display_name: &str,  interval_kind: IntervalKind) -> Result<Box<dyn Function>> {}
}

```

See [cast](https://github.com/datafuselabs/databend/blob/b9ea557c4a060b4622c1287bc7af71e126e609be/common/functions/src/scalars/expressions/cast.rs)

Reply:
make sense